### PR TITLE
Optional Values

### LINKED ISSUE #6: Make app not reliant on export
Add Figaro gem or initialize gem with:

* CLIENT_ID
* CLIENT_SECRET
* URL
* TOKEN_URL

### PR DESCRIPTION
**Pull request by michaelachrisco**
Fixes #6

### CODE REVIEW

**Comment by michaelachrisco** [comment]
Ready for review.

File: lib/gman_client.rb [hunk 1]
```diff
@@ -5,12 +5,16 @@
 
 # GmanClient
 module GmanClient
-  @url = ENV['WRAP_URL']
-  @token_url = ENV['TOKEN_URL']
-  @client_id = ENV['CLIENT_ID']
-  @client_secret = ENV['CLIENT_SECRET']
-  class << self
+  # GC
+  class GC
```
Comment (josephbridgwaterrowe) [inline lib/gman_client.rb]: What a name of `Client` be more intuitive here?

File: spec/lib/driver_commission_history_response_spec.rb [hunk 2]
```diff
@@ -6,10 +6,22 @@
 require 'support/vcr'
 
 RSpec.describe GmanClient do
+  let(:url) { 'http://localhost:3000' }
+  let(:token_url) { 'http://localhost:3000/oauth/token' }
+  # rubocop:disable Metrics/LineLength
+  let(:client_id) { 'd137a94543da86e52214d3ed86b015f9299ad2fc66681637600afad814cd7d2b' }
```
Comment (josephbridgwaterrowe) [inline spec/lib/driver_commission_history_response_spec.rb]: I curious why these are needed, VCR?

File: lib/gman_client.rb (hunk 1 above)
Comment (michaelachrisco) [inline lib/gman_client.rb]: Client is reserved.
I could rename this to:
```GmanClient::Connection```
I am open for suggestions. :)

File: spec/lib/driver_commission_history_response_spec.rb (hunk 2 above)
Comment (michaelachrisco) [inline spec/lib/driver_commission_history_response_spec.rb]: Yes if the client_id does match, then VCR gives you an error.

File: lib/gman_client.rb (hunk 1 above)
Comment (josephbridgwaterrowe) [inline lib/gman_client.rb]: Reserved by whom?

File: spec/lib/driver_commission_history_response_spec.rb (hunk 2 above)
Comment (josephbridgwaterrowe) [inline spec/lib/driver_commission_history_response_spec.rb]: Doesn't? Is this the client_id in the cassettes (or the id used to make the original request the cassette is based upon)?

Comment (michaelachrisco) [inline spec/lib/driver_commission_history_response_spec.rb]: If you look at 
https://github.com/westernmilling/gman_client/blob/master/spec/vcr_cassettes/driver_commission_histories.yml#L8
You can see that the grant type uses client_id and client_secret.

File: lib/gman_client.rb (hunk 1 above)
Comment (michaelachrisco) [inline lib/gman_client.rb]: I was mistaken. Changing now. I believe my first version used a gem with Client reserved and that added to my confusion. Thanks!

File: spec/lib/driver_commission_history_response_spec.rb (hunk 2 above)
Comment (josephbridgwaterrowe) [inline spec/lib/driver_commission_history_response_spec.rb]: OK, thanks.

I assume this is a temporary development key that wont expose any sensitive data? :)

Comment (michaelachrisco) [inline spec/lib/driver_commission_history_response_spec.rb]: Yes. In this case, the client_id and secret are preset in the VCR and not in the API. You would have to add these values into the API DB in order to get a response from the client.

Comment (josephbridgwaterrowe) [inline spec/lib/driver_commission_history_response_spec.rb]: Cool.

**Pull request merged**